### PR TITLE
added NOP keygen script & set-node-key cmd

### PR DESCRIPTION
**Pull request by KuphJr**

### CODE REVIEW

File: tasks/Functions-admin/setNodeKey.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    const oracleFactory = await ethers.getContractFactory("FunctionsOracle")
+    const oracle = oracleFactory.attach(networkConfig[network.name]["functionsOracle"])
+
+    const accounts = await ethers.getSigners()
+    const sender = accounts[0]
+
+    const nodeAddress = taskArgs.node ?? sender.address
+
+    console.log(`Setting oracle OCR config for oracle ${networkConfig[network.name]["functionsOracle"]}`)
```
Comment (pinebit) [inline tasks/Functions-admin/setNodeKey.js:28]: Wrong message

File: tasks/Functions-admin/setNodeKey.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    const setTx = overrides
+      ? await oracle.setNodePublicKey(nodeAddress, "0x" + taskArgs.key, overrides)
+      : await oracle.setNodePublicKey(nodeAddress, "0x" + taskArgs.key)
+
+    console.log(`Waiting ${VERIFICATION_BLOCK_CONFIRMATIONS} blocks for transaction ${setTx.hash} to be confirmed...`)
+    await setTx.wait(VERIFICATION_BLOCK_CONFIRMATIONS)
+
+    console.log(`\nOCR2Oracle config set for oracle ${oracle.address} on ${network.name}`)
```
Comment (pinebit) [inline tasks/Functions-admin/setNodeKey.js:36]: Wrong message

**Review by pinebit — CHANGES_REQUESTED** [review]

File: tasks/Functions-admin/setNodeKey.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+task("functions-set-node-key", "Sets the per-node public key in the Functions oracle contract")
+  .addParam("key", "Node-assigned public key")
+  .addOptionalParam("node", "Address of the node for which the public key is to be set (defaults to msg.sender)")
+  .setAction(async (taskArgs) => {
+    if (network.name === "hardhat") {
+      throw Error("This command cannot be used on a local development chain.  Specify a valid network.")
+    }
+
+    let overrides = undefined
+    if (network.name === "goerli") {
```
Comment (bolekk) [inline tasks/Functions-admin/setNodeKey.js:12]: We shouldn't be copy-pasting this block in every command. At least make it a shared functions.

Comment (KuphJr) [reply tasks/Functions-admin/setNodeKey.js:12]: Is this change going to block this PR or can we merge it for now?  Changing this is going to require changing every task file in the repo.

Comment (bolekk) [reply tasks/Functions-admin/setNodeKey.js:12]: I haven't "requested changes" so once @pinebit approves you're good to go.

File: tasks/Functions-admin/setNodeKey.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      overrides = {
+        // be careful, this may drain your balance quickly
+        maxPriorityFeePerGas: ethers.utils.parseUnits("50", "gwei"),
+        maxFeePerGas: ethers.utils.parseUnits("50", "gwei"),
```
Comment (pinebit) [inline tasks/Functions-admin/setNodeKey.js:16]: I would wish this param to be configured globally. Please create a ticket for this.

**Review by pinebit — APPROVED** [review]

**Pull request merged**